src/FrameBuffer.cpp

```diff
@@ -264,12 +265,19 @@ void FrameBuffer::copyRdram()
 			// This is necessary for auxilary buffers: game can restore content of RDRAM when buffer is not needed anymore
 			// Thus content of RDRAM on moment of buffer creation will be the same as when buffer becomes obsolete.
 			// Validity check will see that the RDRAM is the same and thus the buffer is valid, which is false.
-			// It can be enough to write data just little more than treshold level, but more safe to write twice as much in case that some values in buffer match our fingerprint.
 			const u32 twoPercent = dataSize / 200;
 			u32 start = m_startAddress >> 2;
 			u32 * pData = (u32*)RDRAM;
 			for (u32 i = 0; i < twoPercent; ++i)
-				pData[start++] = m_startAddress;
+			{
+				if (i < 4)
+					pData[start++] = fingerprint[i];
+				else
+					pData[start++] = 0;
```

Comment:
It's needed for Link's model in Zelda OOT in the equipment subscreen. This model is modified by the CPU. I'm pretty sure the CPU does some filtering. The CPU can modify the fingerprint and validity check will fail. If I clear the data below the fingerprint the CPU won't touch my fingerprint.

After adding this it also fixed validity checking problems in Pokemon Stadium 2.